cmd/admin.go

```diff
@@ -606,3 +626,43 @@ func runDeleteAuth(c *cli.Context) error {
 
 	return models.DeleteSource(source)
 }
+
+func runSendMail(c *cli.Context) error {
+	if !c.IsSet("title") {
+		return errors.New("--title flag is missing")
+	}
+
+	if !c.IsSet("content") {
+		return errors.New("--content flag is missing")
+	}
+
+	subject := c.String("title")
+	body := c.String("content")
+
+	if err := initDB(); err != nil {
+		return err
+	}
+
+	users, _, err := models.SearchUsers(&models.SearchUserOptions{
```

Comment:
imho this should really use paging

Reply:
Hey @lafriks I may not get your idea completely :disappointed_relieved: ,
do you mean to send a message to each user separately?

Reply:
I'll answer for @lafriks, he means to fetch users 50 at a time because this SQL Query could be expensive on servers that have thousands of users having to save all of that data into memery. Here is a link to an example of paging: https://github.com/go-gitea/gitea/blob/64133126cdb842a047b2cd104c2f177f6e83f79a/models/migrations/v143.go#L21 (I couldn't find one for users, but the general idea remains the same).

Reply:
I tried to have this done but it seems to me that changes has to be done in https://github.com/go-gitea/gitea/blob/c752ccee6425091cf17c98029f039f3fe66552b4/models/user.go#L1660

Reply:
@techknowlogick  @lafriks 

I used a latest `IterateUser` function which uses batching under the hood.

2ed24b5